### PR TITLE
[SD-661] Adds og:image meta property

### PR DESCRIPTION
**Pull request by PSzyszkaSpark**
Fixes https://spark-solutions.atlassian.net/browse/SD-661

### CODE REVIEW

File: core/app/helpers/spree/base_helper.rb [hunk 1]
```diff
@@ -65,6 +65,23 @@ def meta_data
...
     end
 
+    def image_url_path
+      object = instance_variable_get('@' + controller_name.singularize)
+      return unless object.is_a?(Spree::Product)
+
+      image = default_image_for_product_or_variant(object)
+
+      if image
+        ActiveStorage::Blob.service.path_for(image.attachment.key)
```
Comment (damianlegawiec) [inline core/app/helpers/spree/base_helper.rb]:
```suggestion
        main_app.url_for(image.url)
```

File: core/app/helpers/spree/base_helper.rb [hunk 2]
```diff
@@ -65,6 +65,23 @@ def meta_data
...
+      else
+        image_tag(Spree::Config[:logo])[10..image_tag(Spree::Config[:logo]).length - 5]
```
Comment (damianlegawiec) [inline core/app/helpers/spree/base_helper.rb]:
```suggestion
        main_app.url(Spree::Config[:logo])
```

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: core/app/helpers/spree/base_helper.rb [hunk 3]
```diff
@@ -65,6 +65,18 @@ def meta_data
       meta
     end
 
+    def image_url_path
+      object = instance_variable_get('@' + controller_name.singularize)
+      return unless object.is_a?(Spree::Product)
+
+      image = default_image_for_product_or_variant(object)
+      image.nil? ? asset_path(Spree::Config[:logo]) : main_app.url_for(image.url(image))
```
Comment (damianlegawiec) [inline core/app/helpers/spree/base_helper.rb]:
```suggestion
      image.nil? ? asset_path(Spree::Config[:logo]) : main_app.url_for(image.attachment)
```

**Review by damianlegawiec — CHANGES_REQUESTED** [review]
@PSzyszkaSpark also let's add some tests for this

**Review by damianlegawiec — APPROVED** [review]

**Review by damianlegawiec — APPROVED** [review]

File: core/spec/helpers/base_helper_spec.rb [hunk 4]
```diff
@@ -239,4 +239,55 @@ def link_to_tracking_html(options = {})
...
+
+    context 'when object is product' do
+      let!(:product) { build :product }
+      before do
+        allow_any_instance_of(Spree::BaseHelper).to receive(:instance_variable_get).and_return(product)
+      end
+
+      context 'and has no images attached' do
+        it 'returns spree logo url' do
+          expect(image_url_path).to eq asset_path(Spree::Config[:logo])
```
Comment (LemonAndroid) [inline core/spec/helpers/base_helper_spec.rb]: What if the shop is selling Superman and Batman figures and the user googles "Batman" and the shop has a picture of "Superman" as it's `Spree::Config[:logo]`?

Will the user see a picture of "Superman" then if they google "Batman" in google image search?

**Pull request merged**